### PR TITLE
Fixing Manta inversions in varfish_annotator/annotate_sv (#245)

### PR DESCRIPTION
**Pull request by holtgrewe**

### CODE REVIEW

**Comment by holtgrewe** [comment]
@eudesbarbosa this should resolve the issues that we had with DRAGEN

File: snappy_wrappers/wrappers/varfish_annotator/annotate_sv/wrapper.py [hunk 1]
```diff
@@ -59,13 +66,16 @@
 
 samples=$(cut -f 2 {snakemake.input.ped} | tr '\n' ',' | sed -e 's/,$//g')
 
+# Fix the Manta inversions
+python3 {fix_manta_invs} {snakemake.config[static_data_config][reference][path]} {snakemake.input.vcf} $TMPDIR/tmp4.vcf
```
Comment (eudesbarbosa) [inline snappy_wrappers/wrappers/varfish_annotator/annotate_sv/wrapper.py:70]: Is it possible to use the `fix_manta_invs` arguments here for readability? 
Like:
```
python3 {fix_manta_invs} \
              --reference-fasta {snakemake.config[static_data_config][reference][path]} \
              --input-vcf {snakemake.input.vcf} \
              --output-vcf $TMPDIR/tmp4.vcf
```

File: snappy_wrappers/wrappers/varfish_annotator/annotate_sv/wrapper.py [hunk 2]
```diff
@@ -59,13 +66,16 @@
...
+bcftools sort -o $TMPDIR/tmp3.vcf $TMPDIR/tmp4.vcf
```
Comment (eudesbarbosa) [inline snappy_wrappers/wrappers/varfish_annotator/annotate_sv/wrapper.py:71]: Is it possible to use more meaningful names? `tmp*.vcf` is confusing.

File: snappy_wrappers/wrappers/varfish_annotator/environment.yaml [hunk 3]
```diff
@@ -4,6 +4,9 @@ channels:
 dependencies:
   - bcftools >=1.9
   - htslib >=1.9
+  - samtools >=1.9
   - varfish-annotator-cli ==0.27
   - openjdk >=8,<9
   - jq
+  - vcfpy
+  - pysam
```
Comment (eudesbarbosa) [inline snappy_wrappers/wrappers/varfish_annotator/environment.yaml:12]: Do you need `pysam`? It seems like you are using a `subprocess` call.

**Review by eudesbarbosa — COMMENTED** [review]
Looks great, elegant solution. 

My only concern is that there are no tests for `snappy_wrappers/wrappers/varfish_annotator/annotate_sv/fix_manta_invs.py`. 
Is it possible to have at least one for the whole call? It can even use the [DRAGEN manual](https://github.com/Illumina/manta/blob/master/docs/userGuide/README.md#inversions) example as input.

**Comment by holtgrewe** [comment]
@eudesbarbosa I agree that having no tests is bad in principle. I'm adding a ticket for this but we need a solution now. Crafting the tests would also require creating a fake reference sequence...

File: snappy_wrappers/wrappers/varfish_annotator/annotate_sv/wrapper.py (hunk 2 above)
Comment (holtgrewe) [reply snappy_wrappers/wrappers/varfish_annotator/annotate_sv/wrapper.py]: doen

File: snappy_wrappers/wrappers/varfish_annotator/environment.yaml (hunk 3 above)
Comment (holtgrewe) [reply snappy_wrappers/wrappers/varfish_annotator/environment.yaml:12]: vcfpy needs pysam as this is not int he bioconda package dependency

**Comment by holtgrewe** [comment]
@eudesbarbosa should be good to merge

**Review by eudesbarbosa — APPROVED** [review]
Go for it!

**Comment by holtgrewe** [comment]
Hrm, I'm getting a NPE now from VarfishAnnotator. Oh, well.

**Pull request merged**